app/src/main/java/com/flowfoundation/wallet/manager/walletconnect/WalletConnectDelegate.kt

```diff
@@ -17,20 +18,64 @@ import com.flowfoundation.wallet.widgets.webview.evm.dialog.EvmRequestAccountDia
 import com.flowfoundation.wallet.widgets.webview.evm.model.EVMDialogModel
 import com.flowfoundation.wallet.widgets.webview.fcl.dialog.FclAuthnDialog
 import com.flowfoundation.wallet.widgets.webview.fcl.model.FclDialogModel
+import com.reown.android.Core
+import com.reown.android.CoreClient
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.async
 
 private val TAG = WalletConnectDelegate::class.java.simpleName
 
 internal class WalletConnectDelegate : SignClient.WalletDelegate {
 
     private var isConnected = false
     private val processedRequestIds = mutableSetOf<Long>()
+    private var pendingRedirectUrl: String? = null
+    private var isRedirecting = false
+    private var isSessionSettled = false
+    private var isAuthnComplete = false
 
     /**
      * Triggered whenever the connection state is changed
      */
     override fun onConnectionStateChange(state: Sign.Model.ConnectionState) {
         logd(TAG, "onConnectionStateChange() state:${state.isAvailable}")
         isConnected = state.isAvailable
+        if (!state.isAvailable) {
+            logd(TAG, "Connection lost, attempting to reconnect")
+            ioScope {
+                delay(1000) // Wait before reconnecting
+                try {
+                    CoreClient.Relay.connect { error: Core.Model.Error ->
+                        loge(TAG, "CoreClient.Relay connect error: $error")
+                    }
+                } catch (e: Exception) {
+                    loge(TAG, "Error reconnecting: ${e.message}")
+                    loge(e)
+                }
+            }
+        } else if (isRedirecting) {
+            // If we were trying to redirect and connection is restored, try again
+            performRedirect()
+        }
+    }
+
+    private fun performRedirect() {
+        val redirectUrl = pendingRedirectUrl ?: return
+        val activity = BaseActivity.getCurrentActivity() ?: run {
+            loge(TAG, "No current activity found for redirection")
+            return
+        }
+        
+        logd(TAG, "Attempting to redirect to: $redirectUrl")
+        try {
+            redirectUrl.openInSystemBrowser(activity, true)
```

Comment:
The user might not really coming form system browser, for example they could deeplink from telegram to flow wallet. If we navigate them to browser, it's wired.